flink-scala/src/main/scala/org/apache/flink/api/scala/PartitionSortedDataSet.scala

```diff
@@ -39,12 +41,25 @@ class PartitionSortedDataSet[T: ClassTag](set: SortPartitionOperator[T])
     this
   }
 
-/**
- * Appends the given field and order to the sort-partition operator.
- */
+  /**
+   * Appends the given field and order to the sort-partition operator.
+   */
   override def sortPartition(field: String, order: Order): DataSet[T] = {
     this.set.sortPartition(field, order)
     this
   }
 
+  /**
+    * Appends the given field and order to the sort-partition operator.
+    */
+  override def sortPartition[K: TypeInformation](fun: T => K, order: Order): DataSet[T] = {
```

Comment:
Do not allow chaining of `KeySelector`s. Make sure different key types are not mixed.